pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -0,0 +1,119 @@
+// +k8s:openapi-gen=true
+type ScrapeConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata
+	// More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// List of ScrapeConfigs
+	Items []*ScrapeConfig `json:"items"`
+}
+
+// DeepCopyObject implements the runtime.Object interface.
+func (l *ScrapeConfigList) DeepCopyObject() runtime.Object {
+	return l.DeepCopy()
+}
+
+// ScrapeConfigSpec is a specification of the desired configuration for a scrape configuration.
+type ScrapeConfigSpec struct {
+	StaticConfigs []StaticConfig `json:"staticConfigs,omitempty"`
+	FileSDConfigs []FileSDConfig `json:"fileSDConfigs,omitempty"`
+	HTTPSDConfigs []HTTPSDConfig `json:"httpSDConfigs,omitempty"`
+	// RelabelConfigs to apply to samples before scraping.
+	// Prometheus Operator automatically adds relabelings for a few standard Kubernetes fields.
+	// The original scrape job's name is available via the `__tmp_prometheus_job_name` label.
+	// More info: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config
+	RelabelConfigs []*v1.RelabelConfig `json:"relabelings,omitempty"`
+	// MetricsPath HTTP path to scrape for metrics. If empty, Prometheus uses the default value (e.g. /metrics).
+	MetricsPath string `json:"metricsPath,omitempty"`
+}
+
+// StaticConfig defines a prometheus static configuration
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config
+type StaticConfig struct {
+	// List of targets for this scrape configuration
+	Targets []Target `json:"targets"`
+	// List of labels to add to the metrics scrape as part of this configuration
+	// +optional
+	Labels map[v1.LabelName]string `json:"labels,omitempty"`
+}
+
+// FileSDConfig defines a prometheus file service discovery configuration
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#file_sd_config
+type FileSDConfig struct {
+	// List of files to be used for file discovery. Recommendation: use absolute paths. While relative paths work, the
+	// prometheus-operator project can't guarantee that the working directly will stay the same over time.
+	// Files must be mounted using Prometheus.ConfigMaps or Prometheus.Secrets.
+	// +kubebuilder:validation:MinItems:=1
+	Files []string `json:"files"`
+	// Configures the refresh interval for the file. Prometheus will reload the content of the file at the interval
+	// defined by RefreshInterval.
+	// +optional
+	RefreshInterval v1.Duration `json:"refreshInterval,omitempty"`
+}
+
+// HTTPSDConfig defines a prometheus HTTP service discovery configuration
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#http_sd_config
+type HTTPSDConfig struct {
+	// URL to request service discovery from.
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:Pattern:="^http(s)?://.+$"
```

Comment:
Looking at other definitions we have of URL it's not clear 100% to me that we want to have this validation 🤔

Reply:
Then please my comment, I was just trying to see what we were already doing, if this will always hold endpoints with that pattern then it's good!